core/src/fnc/array.rs

```diff
@@ -13,6 +13,20 @@ use crate::sql::array::Uniq;
 use crate::sql::value::Value;
 
 use rand::prelude::SliceRandom;
+use std::mem::size_of_val;
+
+/// Returns an error if an array of this length is too much to allocate.
+fn limit(name: &str, n: usize) -> Result<(), Error> {
+	const LIMIT: usize = 2usize.pow(20);
```

Comment:
Shall we use an environment variable so one can change this value according to their needs?

Reply:
Addressed in [7c981f8](https://github.com/surrealdb/surrealdb/pull/4184/commits/7c981f888bd57d4a58879cc429198d5d69e3153c)